### PR TITLE
Add error when running overlay over btrfs.

### LINKED ISSUE #9820: /dev/shm troubles in docker 1.4.1
This doesn't work:

```
root@callisto ~ $ docker run --rm ubuntu:14.04 date
FATA[0000] Error response from daemon: mkdir /home/docker/overlay/d2dbd3b3186dc7899502af99203459dc083b210c6c6efbb69d01ee871f534565-init/merged/dev/shm: invalid argument
```

But this works:

```
root@callisto ~ $ docker run --rm busybox date
Fri Dec 26 19:19:18 UTC 2014
```

```
root@callisto ~ $ docker info
Containers: 6
Images: 50
Storage Driver: overlay
Execution Driver: native-0.2
Kernel Version: 3.18.1-031801-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 2
Total Memory: 3.791 GiB
Name: callisto
ID: ZWU6:UVBQ:HL5R:AAB5:GKYQ:2MSR:UPQW:TN6H:32C4:JJFX:WLWA:YPNZ
WARNING: No swap limit support
```

cc @alexlarsson

### PR DESCRIPTION
**Pull request by jessfraz**
Fixes #9820.

### CODE REVIEW

File: daemon/graphdriver/overlay/overlay.go [hunk 1]
```diff
@@ -99,6 +99,17 @@ func Init(home string, options []string) (graphdriver.Driver, error) {
 		return nil, graphdriver.ErrNotSupported
 	}
 
+	// check if they are running over btrfs
+	var buf syscall.Statfs_t
+	if err := syscall.Statfs(path.Dir(home), &buf); err != nil {
+		return nil, err
+	}
+
+	if graphdriver.FsMagic(buf.Type) == graphdriver.FsMagicBtrfs {
```
Comment (LK4D4) [inline daemon/graphdriver/overlay/overlay.go]: This is little confusing :) Message is about using over `ext4`, but you checking for `btrfs`. Maybe I am using `reiser`.
Should we deny to run overlay over any fs which is not `ext4`?

Comment (xiaods) [inline daemon/graphdriver/overlay/overlay.go]: @SvenDowideit  mentioned below request:
can we also add the underlying FS to docker info? (for all graph drivers that have them..)

it also need consider.

Comment (jfrazelle) [inline daemon/graphdriver/overlay/overlay.go]: @xiaods I think that can be a separate pr

Comment (jfrazelle) [inline daemon/graphdriver/overlay/overlay.go]: @LK4D4 I will update

**Comment by jfrazelle** [comment]
I got the magic number for ext4 from http://www.cs.fsu.edu/~baker/devices/lxr/http/source/linux/include/linux/magic.h

File: daemon/graphdriver/overlay/overlay.go (hunk 1 above)
Comment (jfrazelle) [inline daemon/graphdriver/overlay/overlay.go]: updated!

**Comment by LK4D4** [comment]
LGTM

**Comment by icecrime** [comment]
*Review session with @tiborvass @crosbymichael @unclejack*

@LK4D4 We're actually disagreeing with your previous comment: we'd rather have a blacklist of *unsupported fs* (btrfs + aufs for the moment) than a whitelist of *supported fs* (especially as ext4 shouldn't be the only one). What about things such as exfs?

**Comment by crosbymichael** [comment]
Overlayfs is supported with XFS

**Comment by jfrazelle** [comment]
ah ok cool so should I just blacklist Aufs and btrfs what about reiser

**Comment by jfrazelle** [comment]
ok I updated, I put two seperate errors, but I would combine into one if, but I was just trying to avoid someone opening an issue about how we don't tell them what they are running and the error is "vague"

**Comment by crosbymichael** [comment]
LGTM
